File: ImportadorFotos_Dinamico_WPF_VB/ConfiguracionApp.vb
' ============================================================================
' Proyecto: ImportadorFotos_Dinamico_WPF_VB
' Fichero: ConfiguracionApp.vb
' Autor original: El Guille (elguillemola.com)
' Estructura de datos para persistencia JSON: Gemini (La IA de Google)
' Fecha: Junio de 2026
' ============================================================================

Public Class ConfiguracionApp
    Public Property DirectorioDestino As String = String.Empty
    Public Property FechaFiltro As Date? = Date.Today
    Public Property ReemplazarExistentes As Boolean
    Public Property UsarDateTaken As Boolean = True
    Public Property TextoSesionGlobal As String = "sesion 1"
    Public Property UltimoPerfilId As String = String.Empty

    ' Lista dinámica de orígenes configurados por el usuario
    Public Property PerfilesOrigen As New List(Of PerfilOrigen)()
End Class

Public Class PerfilOrigen
    Public Property Id As String = String.Empty
    Public Property Ruta As String = String.Empty
    Public Property Plantilla As String = String.Empty
    Public Property UsarSesion As Boolean
End Class

File: ImportadorFotos_Dinamico_WPF/ConfiguracionApp.cs
// ============================================================================
// Proyecto: ImportadorFotos_Dinamico_WPF
// Fichero: ConfiguracionModel.cs
// Autor original: El Guille (elguillemola.com)
// Diseño de arquitectura JSON: Gemini (La IA de Google)
// Fecha: Junio de 2026
// ============================================================================

using System;
using System.Collections.Generic;

namespace ImportadorFotos_Dinamico_WPF
{
    /// <summary>
    /// Representa la estructura global del archivo de configuración JSON.
    /// </summary>
    public class ConfiguracionApp
    {
        public string UltimoPerfilId { get; set; } = string.Empty;


        //public bool CopiarRAWSeparado { get; set; } = true; // Por defecto true

        public string DirectorioDestino { get; set; } = string.Empty;
        public DateTime? FechaFiltro { get; set; } = DateTime.Today;
        public bool ReemplazarExistentes { get; set; }
        public bool UsarDateTaken { get; set; } = true;
        public string TextoSesionGlobal { get; set; } = "sesion 1";

        // Lista dinámica de perfiles u orígenes configurados por el usuario
        public List<PerfilOrigen> PerfilesOrigen { get; set; } = new List<PerfilOrigen>();
    }

    /// <summary>
    /// Representa cada uno de los orígenes de fotos dinámicos (perfiles).
    /// </summary>
    public class PerfilOrigen
    {
        public string Id { get; set; } = string.Empty;       // Ej: "iPhone de Guille"
        public string Ruta { get; set; } = string.Empty;     // Ej: "S:\iPhone"
        public string Plantilla { get; set; } = string.Empty; // Ej: "$MM $dd 1 iP7"
        public bool UsarSesion { get; set; }                // Checkbox individual de sesión
    }
}
